lib/puppet/provider/scheduled_task/win32_taskscheduler.rb

```diff
@@ -239,6 +239,8 @@ def triggers_same?(current_trigger, desired_trigger)
     desired['months']      ||= current_trigger['months']      if current_trigger.has_key?('months')
     desired['on']          ||= current_trigger['on']          if current_trigger.has_key?('on')
     desired['day_of_week'] ||= current_trigger['day_of_week'] if current_trigger.has_key?('day_of_week')
+    desired['minutes_interval'] ||= current_trigger['minutes_interval'] if current_trigger.has_key?('minutes_interval')
+    desired['minutes_duration'] ||= current_trigger['minutes_duration'] if current_trigger.has_key?('minutes_duration')
```

Comment:
I just copied those lines from the above lines (desired['months'] as they seem to have been working fine. In short, the if statement only applies to the current_trigger['minutes_duration'] and not the whole line. It should mean something like:
if desired['minutes_interval'] is unset, make it equal to the current_trigger['minutes_interval'] if the current_trigger['minutes_interval'] exists.